pkg/services/ngalert/remote/alertmanager.go

```diff
@@ -238,30 +267,44 @@ func (am *ExternalAlertmanager) GetReceivers(ctx context.Context) ([]apimodels.R
 	return rcvs, nil
 }
 
-func (am *ExternalAlertmanager) TestReceivers(ctx context.Context, c apimodels.TestReceiversConfigBodyParams) (*notifier.TestReceiversResult, error) {
+func (am *Alertmanager) TestReceivers(ctx context.Context, c apimodels.TestReceiversConfigBodyParams) (*notifier.TestReceiversResult, error) {
 	return &notifier.TestReceiversResult{}, nil
 }
 
-func (am *ExternalAlertmanager) TestTemplate(ctx context.Context, c apimodels.TestTemplatesConfigBodyParams) (*notifier.TestTemplatesResults, error) {
+func (am *Alertmanager) TestTemplate(ctx context.Context, c apimodels.TestTemplatesConfigBodyParams) (*notifier.TestTemplatesResults, error) {
 	return &notifier.TestTemplatesResults{}, nil
 }
 
-func (am *ExternalAlertmanager) StopAndWait() {
+// TODO: change implementation, this is only useful for testing other methods
+func (am *Alertmanager) Ready() bool {
```

Comment:
Quick readiness check, this will be changed for a more definitive implementation later.